### ISSUE TITLE
markers "fly" in on region change

### ISSUE THREAD

**Issue by macnube** [airbnb/react-native-maps#1103]
The behavior that I'm experiencing is very similar to #774 where the markers appear to fly in after a state or region change. Though this doesn't happen every single time as you'll see in the below gif. 

I thought it might be related to #728 but I've updated to the latest and I'm still experiencing the problem:

react-native: 0.42.0
react-native-maps: ^0.13.0

I'm creating my markers as follows in an iterable which gets recalculated on every region change:

        <MapView.Marker
          coordinate={place.latlng}
          anchor={anchor}
          key={index}
          onPress={() => this.props.handleMarkerClick(place)}>
          <MapMarker
            selected={selected}
            place={place}
            showName={this.props.showNames} />
        </MapView.Marker>

Here is the behavior, you'll notice that it doesn't happen on every single region change. 

![iconflying](https://cloud.githubusercontent.com/assets/7122725/23767629/005b3b16-04e8-11e7-9f9a-44561c122a47.gif)

**Comment by macnube** [comment]
Sure enough I may have just fixed this. 

Earlier I was setting the key value of the MapView.Marker to be the current index from the map that it was within. So I think what was happening was that in certain situations, when recalculating the markers on region change, an index was being assigned to a new Marker that was actually already assigned to previous Marker. This would explain why I saw the issue when I was zoomed into the map as I calculated the markers based on function of the visible region.

The solution was to use unique key values based on the place ids returned from Google. That way, regardless of the marker calculations I'm guaranteed to never use the same key over again for two different markers. 

Not sure if this is expected behavior or not so I'll keep the issue open for now.
